### DISCUSSION TITLE
Decklink SDK Version

### DISCUSSION THREAD

**Discussion by mikekoetter** [CESNET/UltraGrid#418] [General]
Any reason the DecklinkSDK has not been updated?  Looks like what is in the repo is 3 years old

**Comment by alatteri** [comment]
Might not matter in regards to the usage within UG, but the newer Decklink Linux driver now properly support UHD 12bit RGB 4:4:4 on output when used with Flame and Resolve.

**Comment by MartinPulec** [comment]
(see also #67)

As @alatteri says - it doesn't matter. The older linked version works with newer drivers without a problem. But it doesn't hold vice versa - linking a new version prohibits use with older drivers (in other words there is a forward SDK version compatibility with the driver version but not backward). So in my opinion the only point to update to newer SDK version is that there is some new feature which we want to use in UltraGrid. AFAIK there is no other advantage.

**Comment by MartinPulec** [comment]
Just FYI, we've recently updates to **v12.8** for better support of DeckLink IP devices (but there are also an improvement for non-IP regarding extended state reporting and thermal monitoring).

I've also uploaded the versioned BMD API here: <https://github.com/MartinPulec/desktopvideo_sdk-api>.

It is primarily for our internal use (tracking the changes in BMD API) but if interested, you can look at the changes &ndash; every API version has individual commit (but the I didn't add all the old ones - because every version must be separately downloaded from BMD). To see the difference from that version 12.8, you can use:

    git clone https://github.com/MartinPulec/desktopvideo_sdk-api
    cd desktopvideo_sdk-api
    git diff :/v12.8

If you find something interesting, feel free to let us know :)
